src/query/phrase_prefix_query/mod.rs

```diff
@@ -6,7 +6,7 @@ pub use phrase_prefix_query::PhrasePrefixQuery;
 pub use phrase_prefix_scorer::PhrasePrefixScorer;
 pub use phrase_prefix_weight::PhrasePrefixWeight;
 
-fn prefix_end(prefix_start: &[u8]) -> Option<Vec<u8>> {
+pub(crate) fn prefix_end(prefix_start: &[u8]) -> Option<Vec<u8>> {
```

Comment:
I don't understand the `u8::MAX` logic here

Reply:
I referred to the [phrasePrefixQuery implementation](https://github.com/quickwit-oss/tantivy/blob/6564e0c467106ef8570a0fc4f469017347e3e8ef/src/query/phrase_prefix_query/phrase_prefix_weight.rs#L85), which also filters the term based on the term value prefix.

Reply:
I believe it attempts to find the next larger prefix. Typically, this involves incrementing the last u8 value by 1. However, there is an edge case to consider when the last u8 value is u8::MAX.